syntax = "proto3";
package service;

import weak "datamodel.proto";

message LogMessage {
  bytes msg = 1;
}

message LogResponse {
  bool success = 1;
}

message QuitMessage {
  bool kill = 1;
}

message QuitResponse {
  bool success = 1;
}

message SwitchPhaseMessage {
  int32 phase = 1;
}

message SwitchPhaseResponse {
  bool success = 1;
}

message PackageRequest {
  string session = 1;
}

message PackageResponse {
  bool success = 1;
  PackageNode packageNode = 2;
}

message StatusMessage {
  bool phase = 1;
  bool switch = 2;
}

message StatusResponse {
  string phase = 1;
  int32 phaseID = 2;
  bool switching = 3;
  string error = 4;
}

service ControlService {
  rpc Log(LogMessage) returns (LogResponse) {}
  rpc Quit(QuitMessage) returns (QuitResponse) {}
  rpc SwitchPhase(SwitchPhaseMessage) returns (SwitchPhaseResponse) {}
  rpc GetPackageNode(PackageRequest) returns (PackageResponse) {}
  rpc Status(StatusMessage) returns (StatusResponse) {}
}